### PR TITLE
Do less work in PackageSpecificWarningProperties classes

### LINKED ISSUE #12678: PackageSpecificWarningProperties classes do redundant collection lookups
`PackageSpecificWarningProperties` classes do redundant collection lookups.

### PR DESCRIPTION
**Pull request by drewnoakes**
## Bug

Fixes: https://github.com/NuGet/Home/issues/12678

Regression? Last working version:

## Description

When adding a NuGetLogCode to these collections, the previous code was performing many redundant collection lookups. This changes these methods to do less work.

## PR Checklist

- [x] PR has a meaningful title
- [ ] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [x] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

**Review by donnie-msft — APPROVED** [review]

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/Logging/PackageSpecificWarningProperties.cs [hunk 1]
```diff
@@ -88,24 +88,21 @@ public static PackageSpecificWarningProperties CreatePackageSpecificWarningPrope
...
         public void Add(NuGetLogCode code, string libraryId, NuGetFramework framework)
         {
-            if (Properties == null)
-            {
-                Properties = new Dictionary<NuGetLogCode, IDictionary<string, ISet<NuGetFramework>>>();
-            }
+            Properties ??= new Dictionary<NuGetLogCode, IDictionary<string, ISet<NuGetFramework>>>();
 
-            if (!Properties.ContainsKey(code))
+            if (!Properties.TryGetValue(code, out IDictionary<string, ISet<NuGetFramework>> frameworksByLibraryId))
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Commands/RestoreCommand/Logging/PackageSpecificWarningProperties.cs:93]: looks like we have a bunch of duplication in these classes. I created a follow up, cause I bet there's more :D https://github.com/NuGet/Home/issues/12676

**Review by nkolev92 — APPROVED** [review]

**Comment by nkolev92** [comment]
can you please create the matching issue?

Thanks!

**Comment by nkolev92** [comment]
I reran the failing builds. It should make the PR mergeable once they pass.

**Comment by drewnoakes** [comment]
> can you please create the matching issue?
> 
> Thanks!

https://github.com/NuGet/Home/issues/12678

**Pull request merged**